internal/datacoord/task_index.go

```diff
@@ -57,7 +68,7 @@ func (it *indexBuildTask) GetNodeID() int64 {
 	return it.nodeID
 }
 
-func (it *indexBuildTask) ResetNodeID() {
+func (it *indexBuildTask) ResetTask(mt *meta) {
```

Comment:
parameter mt is not in use ?

Reply:
yeah, it is only useful for stats task.

Reply:
stats task needs to be mutually exclusive with other compaction, so when doing the stats task, the state needs to be set to compacting. And the compacting state needs to be reset when retrying.